tests/test_source_aws_secrets_manager.py

```diff
@@ -0,0 +1,120 @@
+"""
+Test pydantic_settings.AWSSecretsManagerSettingsSource.
+"""
+
+import json
+import os
+
+import pytest
+
+try:
+    import yaml
+    from moto import mock_aws
+except ImportError:
+    yaml = None
+    mock_aws = None
+
+from pydantic import BaseModel, Field
+
+from pydantic_settings import (
+    AWSSecretsManagerSettingsSource,
+    BaseSettings,
+    PydanticBaseSettingsSource,
+)
+from pydantic_settings.sources.providers.aws import import_aws_secrets_manager
+
+try:
+    aws_secrets_manager = True
+    import_aws_secrets_manager()
+    import boto3
+
+    os.environ['AWS_DEFAULT_REGION'] = os.environ.get('AWS_DEFAULT_REGION', 'us-east-1')
+except ImportError:
+    aws_secrets_manager = False
+
+
+MODULE = 'pydantic_settings.sources'
+
+if not yaml:
+    pytest.skip('PyYAML is not installed', allow_module_level=True)
```

Comment:
Do we need `pyyaml` to be installed for this test?

Reply:
Yes. Moto the mocking library for boto3 has a dependency on responses which imports yaml. The second tests run in ci was failing and tests won't even start.

Hence the import moto is inside try catch after trying to import yaml.

And the skip module is because it fails for the @mock_aws decorator.